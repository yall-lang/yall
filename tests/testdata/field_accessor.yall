(struct Friend [
	name::String
	birthday::Date
	coolness::number ])

(func describe-friend (friend::Friend)
	(println "I have a friend named " (Friend.name friend)))
