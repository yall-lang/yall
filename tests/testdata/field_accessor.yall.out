[
    Expression {
        kind: Item,
        values: [
            Identifier(
                "struct",
            ),
            Identifier(
                "Friend",
            ),
            Expression(
                Expression {
                    kind: List,
                    values: [
                        Identifier(
                            "name",
                        ),
                        Identifier(
                            "birthday",
                        ),
                        Identifier(
                            "coolness",
                        ),
                    ],
                },
            ),
        ],
    },
    Expression {
        kind: Item,
        values: [
            Identifier(
                "func",
            ),
            Identifier(
                "describe-friend",
            ),
            Expression(
                Expression {
                    kind: Item,
                    values: [
                        Identifier(
                            "friend",
                        ),
                    ],
                },
            ),
            Expression(
                Expression {
                    kind: Item,
                    values: [
                        Identifier(
                            "println",
                        ),
                        Text(
                            "I have a friend named ",
                        ),
                        Expression(
                            Expression {
                                kind: Item,
                                values: [
                                    Identifier(
                                        "Friend.name",
                                    ),
                                    Identifier(
                                        "friend",
                                    ),
                                ],
                            },
                        ),
                    ],
                },
            ),
        ],
    },
]
